components/style/values/specified/box.rs

```diff
@@ -7,15 +7,277 @@
 use style_traits::{ParseError, ToCss};
 use values::CustomIdent;
 use values::KeyframesName;
+#[cfg(feature = "servo")]
+use values::computed::Context;
 use values::generics::box_::AnimationIterationCount as GenericAnimationIterationCount;
 use values::generics::box_::VerticalAlign as GenericVerticalAlign;
 use values::specified::{AllowQuirks, Number};
 use values::specified::length::LengthOrPercentage;
 
+#[derive(Clone, Copy, Debug, Eq, Hash, MallocSizeOf, Parse, PartialEq, ToComputedValue, ToCss)]
+#[cfg_attr(feature = "servo", derive(Deserialize, Serialize))]
+/// Defines an element’s display type, which consists of
+/// the two basic qualities of how an element generates boxes
+/// <https://drafts.csswg.org/css-display/#propdef-display>
+pub enum Display {
+    /// <display-outside> `inline` variant
+    Inline,
+    /// <display-outside> `block` variant
+    Block,
+    /// <display-legacy> `inline-block` variant
+    InlineBlock,
+    /// <display-inside> `table` variant
+    Table,
+    /// <display-legacy> `inline-table` variant
+    InlineTable,
+    /// <display-internal> `table-row-group` variant
+    TableRowGroup,
+    /// <display-internal> `table-header-group` variant
+    TableHeaderGroup,
+    /// <display-internal> `table-footer-group` variant
+    TableFooterGroup,
+    /// <display-internal> `table-row` variant
+    TableRow,
+    /// <display-internal> `table-column-group` variant
+    TableColumnGroup,
+    /// <display-internal> `table-column` variant
+    TableColumn,
+    /// <display-internal> `table-cell` variant
+    TableCell,
+    /// <display-internal> `table-caption` variant
+    TableCaption,
+    /// <display-listitem> `list-item` variant
+    ListItem,
+    /// <display-box> `none` variant
+    None,
+    /// <display-inside> `flex` variant
+    Flex,
+    /// <display-legacy> `inline-flex` variant
+    InlineFlex,
+    #[cfg(feature = "gecko")]
+    /// <display-inside> `grid` variant
+    Grid,
+    #[cfg(feature = "gecko")]
+    /// <display-legacy> `inline-grid` variant
+    InlineGrid,
+    #[cfg(feature = "gecko")]
+    /// <display-inside> `ruby` variant
+    Ruby,
+    #[cfg(feature = "gecko")]
+    /// <display-internal> `ruby-base` variant
+    RubyBase,
+    #[cfg(feature = "gecko")]
+    /// <display-internal> `ruby-base-container` variant
+    RubyBaseContainer,
+    #[cfg(feature = "gecko")]
+    /// <display-internal> `ruby-text` variant
+    RubyText,
+    #[cfg(feature = "gecko")]
+    /// <display-internal> `ruby-text-container` variant
+    RubyTextContainer,
+    #[cfg(feature = "gecko")]
+    /// <display-box> `contents` variant
+    Contents,
+    #[cfg(feature = "gecko")]
+    /// <display-inside> `flow` variant
+    FlowRoot,
+    #[cfg(feature = "gecko")]
+    /// `box` variant for `webkit`
+    WebkitBox,
+    #[cfg(feature = "gecko")]
+    /// `inline-box` variant for `webkit`
+    WebkitInlineBox,
+    #[cfg(feature = "gecko")]
+    /// `box` variant for `moz`
+    MozBox,
```

Comment:
I'd just `#[allow(missing_docs)]`, for the variants, and remove the comments, they are more noise than good I think, and for keywords that's just fine. In particular, these for example are wrong (`-moz-box` is a XUL display type, as lots of the ones below starting with `-moz`).

Reply:
Using `#[allow(missing_docs)]` makes this `enum` more clear :P